src/transformers/models/llava/modeling_llava.py

```diff
@@ -431,8 +431,11 @@ def forward(
                 if past_key_values is not None and pixel_values is not None and input_ids.shape[1] == 1:
                     # Retrieve the first layer to inspect the logits and mask out the hidden states
                     # that are set to 0
-                    first_layer_past_key_value = past_key_values[0][0][:, 0, :, 0]
-                    batch_index, non_attended_tokens = torch.where(first_layer_past_key_value == 0)
+                    first_layer_past_key_value = past_key_values[0][0][:, :, :, 0]
```

Comment:
A more robust check is to check the entire row that has a dimension of 128 (head_dim) instead of looking only at a single logit value, which can randomly hapoen in some cases in fp16

Reply:
cool!